spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -1,78 +1,296 @@
-import spikeextractors as se
-
 import os
-import numpy as np
-from datetime import datetime
 import uuid
+from datetime import datetime
+
+import numpy as np
+from hdmf.data_utils import DataChunkIterator
+
+import spikeextractors as se
 
 try:
     from pynwb import NWBHDF5IO
     from pynwb import NWBFile
     from pynwb.ecephys import ElectricalSeries
+    from pynwb.ecephys import ElectrodeGroup
+    from pynwb.device import Device
+    from pynwb.misc import Units
+
     HAVE_NWB = True
 except ModuleNotFoundError:
     HAVE_NWB = False
 
-class CopyRecordingExtractor(se.RecordingExtractor):
-    def __init__(self, other):
-        se.RecordingExtractor.__init__(self)
-        self._other = other
-        self.copy_channel_properties(other)
 
-    def get_channel_ids(self):
-        return list(range(self._other.get_num_channels()))
+def check_nwb_install():
+    assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
 
-    def get_num_frames(self):
-        return self._other.get_num_frames()
 
-    def get_sampling_frequency(self):
-        return self._other.get_sampling_frequency()
+def set_dynamic_table_property(dynamic_table, row_ids, property_name, values, default_value=np.nan,
+                               description='no description'):
+    check_nwb_install()
+    if not isinstance(row_ids, list) or not all(isinstance(x, int) for x in row_ids):
+        raise TypeError("'ids' must be an integer or a list of integers")
+    ids = list(dynamic_table.id[:])
+    if any([i not in ids for i in row_ids]):
+        raise ValueError("'ids' contains values outside the range of existing ids")
+    if not isinstance(property_name, str):
+        raise TypeError("'property_name' must be a string")
+    if len(row_ids) != len(values):
+        raise ValueError("'ids' and 'values' should be lists of same size")
 
-    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
-        return self._other.get_traces(channel_ids=channel_ids, start_frame=start_frame, end_frame=end_frame)
+    if property_name in dynamic_table:
+        for (row_id, value) in zip(row_ids, values):
+            dynamic_table[property_name][ids.index(row_id)] = value
```

Comment:
When I try: `recording.set_channel_property(1, 'group', 0)` I get the following error at this line:
`TypeError: 'VectorData' object does not support item assignment`

Reply:
thanks for pointing that out! it should be fixed in the coming PR update